### DISCUSSION TITLE
napi can update ast tree or write origin code ?

### DISCUSSION THREAD

**Discussion by zhousg** [ast-grep/ast-grep#800] [Q&A]
napi can update ast tree or write origin code

**Comment by HerringtonDarkholme** [comment]
Hi, napi cannot update the ast-tree at the moment. You can use magic string instead.
Example
https://github.com/swc-project/swc/blob/4983b1cef26033d0a58132ca7f93c4eb56415ccd/packages/helpers/scripts/ast_grep.js#L2

**Comment by zhousg** [comment]
Thanks
